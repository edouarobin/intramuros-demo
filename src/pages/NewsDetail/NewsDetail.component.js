// @flow

import React, { PureComponent } from 'react';
import {
  StyleSheet,
  View,
  Text,
  Image,
  ScrollView,
  StatusBar,
  TouchableOpacity,
} from 'react-native';

import {
  GoBackButton,
  ContactWeb,
  Schedule,
  Price,
  ImageZoomModal,
  OrganizerCard,
} from 'Intramuros/src/components';
import globalStyle, {
  newsDetailImageHeight,
  STATUS_BAR_HEIGHT,
  BIG_SCREEN_PADDING_OFFSET,
} from 'Intramuros/src/style/globalStyle';
import { formatDateWOweekday } from 'Intramuros/src/services/FormatDate';
import { MyFastImage } from 'Intramuros/src/components';
import HTMLText from 'Intramuros/src/utils/HTMLText';

type PropsType = {
  navigation: NavigationPropsType<>,
  newsDetails: NewsType,
  isNotification: boolean,
  notificationTitle?: string,
  notificationBody: ?string,
  notificationImage?: string,
  notificationCityName?: string,
  notificationCityId?: string,
  notificationAssoId?: string,
  notificationAssoName?: string,
  notificationAssoPicto?: string,
  notificationAssoCity?: string,
};

const styles = StyleSheet.create({
  titleContainer: {
    marginHorizontal: 16,
    alignItems: 'center',
    marginTop: 23,
  },
  titleStyle: {
    fontFamily: 'Lato-Bold',
    fontSize: 20,
    // fontWeight: 'bold',
    color: 'black',
    lineHeight: 28,
    textAlign: 'center',
  },
  dateStyle: {
    fontFamily: 'Lato-Regular',
    marginTop: 8,
    fontSize: 14,
    color: globalStyle.colors.greyishBrown,
    textAlign: 'center',
  },
  seperator: {
    marginTop: 12,
    backgroundColor: globalStyle.colors.yellow,
    height: 2,
    width: 96,
    borderRadius: 1,
  },
  image: {
    flex: 1,
    marginTop: 23,
    width: '100%',
    height: newsDetailImageHeight + STATUS_BAR_HEIGHT,
  },
  descriptionContainer: {
    flex: 1,
    padding: 16,
    paddingHorizontal: 16 + BIG_SCREEN_PADDING_OFFSET,
  },
  decriptionStyle: {
    fontFamily: 'Lato-Regular',
    fontSize: 14,
    lineHeight: 24,
    color: 'black',
  },
  linksStyle: {
    marginTop: 16,
  },
  moreInfo: {
    fontFamily: 'Lato-BoldItalic',
    // fontStyle: 'italic',
    // fontWeight: 'bold',
    color: globalStyle.colors.greyishBrown,
    marginTop: 10,
    marginBottom: 5,
  },
  authorStyle: {
    fontFamily: 'Lato-Italic',
    marginTop: 16,
    fontSize: 14,
    lineHeight: 24,
    color: 'black',
    // fontStyle: 'italic',
  },
});

export default class NewsDetail extends PureComponent<PropsType> {
  state = {
    modalVisible: false,
  };

  openZoomImage = () => {
    this.setState({ modalVisible: true });
  };
  closeImageModal = () => {
    this.setState({ modalVisible: false });
  };

  render() {
    let {
      isNotification,
      newsDetails,
      notificationImage,
      notificationTitle,
      notificationBody,
      notificationCityName,
      notificationAssoId,
      notificationAssoName,
      notificationAssoPicto,
      notificationAssoCity,
    } = this.props;
    let details = {
      title: '',
      city: '',
      date: '',
      image: '',
      description: '',
      schedule: '',
      price: '',
      url1: '',
      url2: '',
      url3: '',
      author: '',
      organizer: '',
      organizer_city: null,
      organizer_name: '',
      organizer_logo: '',
      organizer_type: '',
    };
    //Notification or News
    if (isNotification) {
      details.title = notificationTitle;
      details.city = notificationCityName;
      details.description = notificationBody;
      details.image = notificationImage;
      details.organizer = notificationAssoId;
      details.organizer_city = notificationAssoCity;
      details.organizer_name = notificationAssoName;
      details.organizer_logo = notificationAssoPicto;
      details.organizer_type = 'ASSO';
    } else if (newsDetails) {
      details.title = newsDetails.title;
      details.description = newsDetails.description;
      details.image = newsDetails.image;
      details.date = newsDetails.updatedAt;
      details.schedule = newsDetails.schedule;
      details.price = newsDetails.price;
      details.url1 = newsDetails.url1;
      details.url2 = newsDetails.url2;
      details.url3 = newsDetails.url3;
      details.author = newsDetails.author;
      details.organizer = newsDetails.organizer;
      details.organizer_city = newsDetails.organizer_city;
      details.organizer_name = newsDetails.organizer_name;
      details.organizer_logo = newsDetails.organizer_logo
        ? newsDetails.organizer_logo
        : newsDetails.organizer_category_picto;
      details.organizer_type = newsDetails.organizer_type;
    }

    return (
      <View style={{ flex: 1, backgroundColor: 'white' }}>
        <GoBackButton
          navigation={this.props.navigation}
          color={globalStyle.colors.mainBlue}
        />
        <StatusBar
          barStyle="dark-content"
          translucent
          backgroundColor="transparent"
          animated={true}
        />

        <ScrollView>
          <View style={styles.titleContainer}>
            <Text style={styles.titleStyle}>{details.title}</Text>
            {details.city ? (
              <Text style={styles.dateStyle}>{details.city}</Text>
            ) : null}
            {details.date ? (
              <Text style={styles.dateStyle}>
                Publié le {formatDateWOweekday(details.date)}
              </Text>
            ) : null}
            <View style={styles.seperator} />
          </View>
          {details.image ? (
            <TouchableOpacity onPress={this.openZoomImage}>
              <MyFastImage
                style={styles.image}
                source={{
                  uri: details.image,
                  priority: 'normal',
                }}
                resizeMode={'contain'}
              />
            </TouchableOpacity>
          ) : null}
          <View style={styles.descriptionContainer}>
            <HTMLText
              style={styles.decriptionStyle}
              value={details.description ? details.description : ''}
            />
            {details.schedule ? <Schedule text={details.schedule} /> : null}
            {details.price ? <Price price={details.price} /> : null}
            {details.url1 || details.url2 || details.url3 ? (
              <View style={styles.linksStyle}>
                <Text style={styles.moreInfo}>Plus d'informations :</Text>
                {details.url1 ? (
                  <ContactWeb website={details.url1} fisrtElement={true} />
                ) : null}
                {details.url2 ? <ContactWeb website={details.url2} /> : null}
                {details.url3 ? <ContactWeb website={details.url3} /> : null}
              </View>
            ) : null}
            {details.author ? (
              <Text style={styles.authorStyle}>
                Publié par {details.author}
              </Text>
            ) : (
              <View />
            )}

            {details.organizer ? (
              <OrganizerCard
                title={details.organizer_name}
                logo={details.organizer_logo}
                onPress={() => {
                  if (details.organizer_type === 'ASSO')
                    this.props.navigation.navigate('AssoDetail', {
                      assoId: details.organizer,
                      cityOfAsso: details.organizer_city,
                    });
                  if (details.organizer_type === 'COMMERCE')
                    this.props.navigation.navigate('CommerceDetail', {
                      commerceId: details.organizer,
                    });
                }}
              />
            ) : null}

            <View style={{ height: 60 }} />
          </View>
        </ScrollView>
        {details.image ? (
          <ImageZoomModal
            image={details.image}
            modalVisible={this.state.modalVisible}
            closeImageModal={this.closeImageModal}
          />
        ) : null}
      </View>
    );
  }
}
