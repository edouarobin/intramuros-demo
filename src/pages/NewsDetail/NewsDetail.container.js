// @flow
import { connect } from 'react-redux';

import { selectors } from 'Intramuros/src/redux/news';

import NewsDetail from './NewsDetail.component';

const mapStateToProps = (rootState: RootStateType, ownProps: any) => ({
  newsDetails: selectors.selectNewsDetailsFromID(
    rootState,
    ownProps.navigation.state.params.newsID
  ),
  notificationTitle: ownProps.navigation.state.params.notificationTitle,
  notificationBody: ownProps.navigation.state.params.notificationBody,
  notificationImage: ownProps.navigation.state.params.notificationImage,
  notificationCityName: ownProps.navigation.state.params.notificationCityName,
  notificationCityId: ownProps.navigation.state.params.notificationCityId,
  notificationAssoId: ownProps.navigation.state.params.notificationAssoId,
  notificationAssoName: ownProps.navigation.state.params.notificationAssoName,
  notificationAssoPicto: ownProps.navigation.state.params.notificationAssoPicto,
  notificationAssoCity: ownProps.navigation.state.params.notificationAssoCity,
  isNotification: ownProps.navigation.state.params.isNotification,
});

const mapDispatchToProps = {};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NewsDetail);
