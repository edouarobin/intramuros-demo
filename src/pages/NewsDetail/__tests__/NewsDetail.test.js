// @flow-weak

import React from 'react';
import renderer from 'react-test-renderer';

import NewsDetail from '../NewsDetail.component';

it('Fully renders the NewsDetails page', () => {
  const mockNewsDetails = {
    title: 'News !',
    description: 'dflkjs lkjsdl kfjsdlkj sdlk efef fsf ds f',
    createdAt: '2017-11-08T15:20:33.457740Z',
    updatedAt: '2017-11-14T10:41:36.111827Z',
    city: 5,
    image:
      'https://intramuros.cloud.bam.tech/media/news/673243e4d9232489945e19bff3ec85e9_News__dflkjs_lkj.png',
    id: 1,
  };
  const tree = renderer
    .create(<NewsDetail navigation={{}} newsDetails={mockNewsDetails} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
