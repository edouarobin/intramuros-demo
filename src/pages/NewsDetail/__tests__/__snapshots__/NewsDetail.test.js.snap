// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fully renders the NewsDetails page 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginHorizontal": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "bold",
              "lineHeight": 28,
              "textAlign": "center",
            }
          }
        >
          News !
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#535353",
              "fontSize": 14,
              "marginTop": 8,
              "textAlign": "center",
            }
          }
        >
          Publi√© le 
          14 novembre
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#FFD153",
              "borderRadius": 1,
              "height": 2,
              "marginTop": 12,
              "width": 96,
            }
          }
        />
      </View>
      <Image
        source={
          Object {
            "uri": "https://intramuros.cloud.bam.tech/media/news/673243e4d9232489945e19bff3ec85e9_News__dflkjs_lkj.png",
          }
        }
        style={
          Object {
            "height": 395.52000000000004,
            "marginTop": 23,
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "padding": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontSize": 14,
              "lineHeight": 24,
            }
          }
        >
          dflkjs lkjsdl kfjsdlkj sdlk efef fsf ds f
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
